distributed/sizeof.py

```diff
@@ -1,17 +1,22 @@
 import logging
+from typing import SupportsInt
 
 from dask.sizeof import sizeof
+from dask.utils import format_bytes
 
 logger = logging.getLogger(__name__)
 
 
-def safe_sizeof(obj, default_size=1e6):
+def safe_sizeof(obj, default_size: SupportsInt = 1e6) -> int:
```

Comment:
cc @crusaderky for thoughts on annotating `default_size` here. mypy complained about using `numbers.Number` and `number.Integral`. Maybe we should prefer an explicit `int | float` instead?

Reply:
I took a look at this, since I've wondered the same thing before:

1. `numbers.Integral` fails because the default of `1.e6` is a float.
2. Annotations for builtin numeric types appear to be a bit of a mess right now! See [this long running issue](https://github.com/python/mypy/issues/3186), in which GVR calls the class hierarchy for `numbers` a "flight of fancy" and not "useful"
3. I think ideally we would want `numbers.Real` to work for `int | float`, but it doesn't work as discussed above (and in the many linked issues from that).
4. In [PEP 484](https://www.python.org/dev/peps/pep-0484/#the-numeric-tower) they explicitly recommend just using `float` for cases that accept float-or-int. So I think that would be a fine choice here.
5. But also, I kind of like your choice of `SupportsInt`! It follows the best-practice principle of annotating with the most general annotation available

Reply:
``default_size: float`` is all you need. Everything else is overkill and an unnecessary headache IMHO.